ts/scene_io.ts

```diff
@@ -0,0 +1,55 @@
+import { readFile, writeFile } from 'fs';
+
+import { Scene } from './proto/scene';
+
+export type SceneIO = {
+
+    decode: (data: string | Buffer) => Scene,
+    read: (path: string) => Promise<Scene>,
+
+    encode: (scene: Scene) => Uint8Array,
+    save: (scene: Scene, path: string) => Promise<void>,
+    encodeAsJson: (scene: Scene) => string,
+    saveAsJson: (scene: Scene, path: string) => Promise<void>,
+};
+
+const decode = (data: Buffer | string) => {
+    try {
+        return Scene.fromJSON(JSON.parse(data.toString()));
+    } catch (e) {
+        if (Buffer.isBuffer(data)) {
+            return Scene.decode(data);
+        } else {
+            throw Error('Failed to decode as scene format: input is string, but not a valid json');
+        }
+    }
+};
+
+const encode = (scene: Scene) => Scene.encode(scene).finish();
+const encodeAsJson = (scene: Scene) => JSON.stringify(Scene.toJSON(scene), null, 4);
+
+export const SceneFormatIO : SceneIO = {
+
+    decode,
+    read: async (path: string) => new Promise((resolve, reject) => {
+        readFile(path, (err, data) => {
```

Comment:
Well, this looks good, but later we'd better rewrite this to async/await(optionally) and make file read in a stream so that the raw data is not loaded entirely to memory.(This may slow down the code or even invoke a js error, which our team currently has when parsing a big obj file).

Reply:
I think we will implement stream reading later if we face similar problems. I think we should handle big files fine if those will use binary protobuf.